File: runtime/js/core/browser.js
/*
 _       __      _ __    
| |     / /___ _(_) /____
| | /| / / __ `/ / / ___/
| |/ |/ / /_/ / / (__  ) 
|__/|__/\__,_/_/_/____/  
The lightweight framework for web-like apps
(c) Lea Anthony 2019-present
*/
/* jshint esversion: 6 */

import { SystemCall } from './calls';

/**
 * Opens the given URL in the system browser
 *
 * @export
 * @param {string} url
 * @returns
 */
export function OpenURL(url) {
	return SystemCall('Browser.OpenURL', url);
}

/**
 * Opens the given filename using the system's default file handler
 *
 * @export
 * @param {string} filename
 * @returns
 */
export function OpenFile(filename) {
	return SystemCall('Browser.OpenFile', filename);
}

File: v3/internal/browser/browser.go
// Package browser provides functions to open URLs and files in the default browser.
package browser

// OpenURL opens the named URL in the default browser.
func OpenURL(url string) error {
	return open(url)
}

// OpenFile opens the named file in the default browser or file handler.
func OpenFile(path string) error {
	return open(path)
}
